holoviews/element/raster.py

```diff
@@ -677,13 +678,17 @@ def trimesh(self):
         t3 = np.concatenate([t3s, t4s])
         ts = (t1, t2, t3)
         for vd in self.vdims:
-            zs = self.dimension_values(2)
+            zs = self.dimension_values(vd)
             ts = ts + (np.concatenate([zs, zs]),)
 
         # Construct TriMesh
-        params = {k: v for k, v in util.get_param_values(self).items()
-                  if k != 'kdims'}
-        return TriMesh(((ts,), vertices), **params)
+        params = util.get_param_values(self)
+        params['kdims'] = params['kdims'] + TriMesh.node_type.kdims[2:]
+        node_params = {k: v for k, v in params.items() if k != 'vdims'}
+        nodes = TriMesh.node_type(vertices+(np.arange(len(vertices[0])),),
+                                  **node_params)
+        params = {k: v for k, v in params.items() if k != 'kdims'}
```

Comment:
Very minor point: I would inline this dictionary comprehension to avoid overriding the ``params`` variable again with something derived from itself.

Reply:
You could also do that to avoid defining ``node_params``...